### ISSUE TITLE
about CAPS_LOCK key can't work

### ISSUE THREAD

**Issue by joyspce** [Hammerspoon/hammerspoon#1362]
-- when i run this code,  it work nothing, the capslock light no change
    hs.eventtap.event.newSystemKeyEvent( 'CAPS_LOCK', true):post();
    hs.eventtap.event.newSystemKeyEvent( 'CAPS_LOCK', false):post();

-- when run that code, the capslock light give a flash, a little spark, but hammerspoon console show a error message and then stoped
    -- eventType:FlagsChanged    code:0x39     name:caps_lock   flags:Caps  misc:characters:    
    local  newKey = hs.eventtap.event.newSystemKeyEvent( 0x39, true):post();
    hs.timer.doAfter(0.5, function() hs.eventtap.event.newSystemKeyEvent( 0x39, false):post(); end)

--hammerspoon console show the error message

-- 2017-04-25 04:44:28: *** ERROR: /Users/joys/.hammerspoon/init.lua:92: ERROR: incorrect type 'number' for argument 1 (expected string)
-- stack traceback:
--     [C]: in function 'hs.eventtap.event.newSystemKeyEvent'
--     /Users/joys/.hammerspoon/init.lua:92: in function </Users/joys/.hammerspoon/init.lua:69>
--     /Users/joys/.hammerspoon/init.lua:97: in main chunk
--     [C]: in function 'xpcall'
--     ...app/Contents/Resources/extensions/hs/_coresetup/init.lua:481: in function 'hs._coresetup.setup'
--     (...tail calls...)

this is my mac osX oc define

    -- /System/Library/Frameworks/IOKit.framework/Headers/hidsystem/ev_keymap.h
    
    -- #define NX_NOSPECIALKEY         0xFFFF
    -- #define NX_KEYTYPE_SOUND_UP     0
    -- #define NX_KEYTYPE_SOUND_DOWN       1
    -- #define NX_KEYTYPE_BRIGHTNESS_UP    2
    -- #define NX_KEYTYPE_BRIGHTNESS_DOWN  3
    -- #define NX_KEYTYPE_CAPS_LOCK        4
    -- #define NX_KEYTYPE_HELP         5
    -- #define NX_POWER_KEY            6
    -- #define NX_KEYTYPE_MUTE         7
    -- #define NX_UP_ARROW_KEY         8
    -- #define NX_DOWN_ARROW_KEY       9
    -- #define NX_KEYTYPE_NUM_LOCK     10

**Comment by cmsj** [comment]
I think the LED has to be controlled separately via IOKit, which isn't something we currently support.

@asmagill - I think you added the `newSystemKeyEvent()` - should we be expecting the CapsLock one to actually even enable caps lock?

**Comment by asmagill** [comment]
I don't recall if I *ever* got the Caps lock key to actually change state... I have been able to adjust brightness/volume with these (though I admit I haven't tested even that since 10.9 or 10.10).

I'd like to be able to toggle Caps lock, but the obvious approaches haven't worked and a quick google search didn't suggest anything so it's been on the back burner.

I have successfully detected the Caps lock when it's physically pressed, but even that got weird when I have multiple keyboards attached and went back and forth too quickly...

The original impetus behind adding support for these system keys was to provide a way to detect/invoke the brightness keys on my laptop.  The others looked like they worked the same way, but I admit it was never tested (by me at least) with much depth.
